### ISSUE TITLE
French Cleat edge stud can clash with cleat

### ISSUE THREAD

**Issue by AlexCPU** [florianfesti/boxes#731]
**Describe the bug**
French Cleat settings don't ask for the height of the cleat, so can generate the stud where there isn't space on the wall

**Screenshots or Drawings**
![image](https://github.com/user-attachments/assets/d00a4c80-4e06-4f1a-99fc-56b1cc7bba48)

**To Reproduce**
The French Cleat Settings requests the depth (green in fig.), and the spacing (purple in fig.), but doesn't request the height (blue in fig.) of the cleats, so this means that in stud mode (the default and likely most common), studs can be created in studs that they don't fit.

**Expected behaviour**
For holders smaller than a cleat, omit the stud.
For holders bigger than a cleat, move the stud up if it would collide with a cleat.

**Additional context**
I'll have a go at fixing this soon, but I've got a couple of generators I want to try first.

**Comment by florianfesti** [comment]
Yeah, the stud implementation is pretty simple - not to say brain dead. Probably result of me not having an actual French cleat wall. A patch is very welcome!
